include/nlohmann/detail/input/input_adapters.hpp

```diff
@@ -48,9 +50,14 @@ class file_input_adapter
     using char_type = char;
 
     JSON_HEDLEY_NON_NULL(2)
-    explicit file_input_adapter(std::FILE* f) noexcept
+    explicit file_input_adapter(std::FILE* f)
         : m_file(f)
-    {}
+    {
+        if (m_file == nullptr)
+        {
+            JSON_THROW(parse_error::create(116, 0, detail::concat("input file is invalid: ", reinterpret_cast<const char*>(std::strerror(errno))), nullptr));
```

Comment:
I would not rely on `errno` here. We don't know if another function that sets `errno` was called after `fopen`.

Reply:
Right.